### ISSUE TITLE
Merge in templates instantly on project loading

### ISSUE THREAD

**Issue by joostfarla** [serverless/serverless#1134]
It would be much cleaner to merge in templates (not variables) in the project loading phase. Currently (in v0.5) this causes issues with plugins relying on populated templates. In my case (CORS plugin) I’m forced to use `toObjectPopulated()` to merge in templates AND variables (where I just need the populated templates), which causes issues with non-existing variables. This typically happens when the hook is fired before the user is prompted or before certain variables are determined automatically in a later action. Furthermore, it's probably a bad practice in general to let plugins populate all environment variables.

This issue is discussed in https://github.com/joostfarla/serverless-cors-plugin/issues/19

To summarize the proposal:
* Merge in templates instantly on project loading
* Keep using `toObjectPopulated()` to just populate variables

**Comment by flomotlik** [comment]
As discussed in the other issue this is changed a lot in V1, would be great to chat about those changes and make sure you'r plugins work for V1
